### ISSUE TITLE
Maybe replace 0xfffff with -1 ?

### ISSUE THREAD

**Issue by trufae** [wargio/r2dec-js#30]
in cases like this, were there's no binary operation involved like AND , .. makes sense to convert them to -1
```
    while (eax == 0xffffffffffffffff) {
```

**Comment by Maijin** [comment]
+1

**Comment by elicn** [comment]
TL;DR: this is an r2 bug, rather than r2dec's

For the first occurrence, a bit-wise operation, I would like to see the actual bit-mask rather than a magic value. As for the two comparisons, r2dec recognizes the magical `-1` value according to the underlying architecture, so I looked in the attached binary and noticed that the whole binary is in fact in 32 bits. That made me think -- _what an immediate value of `0xffffffffffffffff` is doing there in the first place_ ?

r2 interprets `83 78 70` as `cmp eax, 0xffffffffffffffff` although the `83` opcode is there to compare registers with an 8-bit immediate value. It is easy to see that this bug came out as a result of sign extension: try to replace the last byte of this sequence with 70h (MSB is clear), and then back to 80h to see the problem.

**Comment by elicn** [comment]
The bottom line is, that r2 should have generated `cmp eax, 0xffffffff`. In that case, the disassembled output would show `-1`.

**Comment by elicn** [comment]
I'll open a ticket for r2

**Comment by radare** [comment]
Blame capstone, not r2
